Content.Server/Store/Conditions/StoreAntagCondition.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Server.Mind.Components;
+using Content.Server.Traitor;
+using Content.Shared.Roles;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Set;
+
+namespace Content.Shared.Store.Conditions;
+
+/// <summary>
+/// Allows a store entry to be filtered out based on the user's antag role.
+/// Supports both blacklists and whitelists. This is copypaste because roles
+/// are absolute shitcode. Refactor this later. -emo
+/// </summary>
+public sealed class StoreAntagCondition : ListingCondition
```

Comment:
Fields and functions of this need doc comments. Remember `<inheritdoc/>` exists for overrides.